entity_event/models.py

```diff
@@ -972,19 +959,24 @@ class Event(models.Model):
 
     objects = EventManager()
 
-    def get_context(self):
-        """Retrieves and populates the context for this event.
+    def __init__(self, *args, **kwargs):
+        super(Event, self).__init__(*args, **kwargs)
+        # A dictionary that is populated with renderers after the contexts have been
+        # properly loaded. When renderers are available, the 'render' method may be
+        # called with a medium and optional observer
+        self._context_renderers = {}
 
-        At the minimum, whatever context was stored in the event is
-        returned. If the source of the event provides a
-        ``context_loader``, any additional context created by that
-        function will be included.
-
-        :rtype: Dict
-        :returns: A dictionary of the event's context, with any
-            additional context loaded.
+    def render(self, medium, observing_entity=None):
+        """
+        Returns the rendered event as a tuple of text and html content. This information
+        is filled out with respect to which medium is rendering the event, what context
+        renderers are available with the prefetched context, and which optional entity
+        may be observing the rendered event.
         """
```

Comment:
observing_entity is variable that will be passed through to the context so that contexts may be rendered with respect to the viewer. I'm also going to add the ability to pass more information into the context if needed later. For now these variables aren't used